### PR TITLE
content: podfoods relevance blog

### PR DESCRIPTION
**Pull request by skeptrunedev**
- wip: podfoods blog
- initial transformation
- removed extraneaous images

### CODE REVIEW

File: src/content/post/podfoods-relevance-blog.mdx [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+    The demo also shows two other search types:
+    <strong>semantic</strong>, a dense vector search using cosine distance vectors (here using{' '}
+    <a href="https://jina.ai/news/jina-ai-launches-worlds-first-open-source-8k-text-embedding-rivaling-openai">
+      Jina english embeddings
+    </a>
+    ), and <strong>hybrid</strong>, which takes in results from both fulltext and hybrid and re-ranks with a
+    cross-encoder model (here the
+    <a href="https://huggingface.co/BAAI/bge-reranker-large">BAAI/bge-reranker-large</a>).
+  </p>
+</div>
```
Comment (skeptrunedev) [inline src/content/post/podfoods-relevance-blog.mdx]: This should use a component with tailwind styling and look correct in dark mode

File: src/content/post/podfoods-relevance-blog.mdx [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+    <a href="https://podfoods.trieve.ai/?term=beautiful+toaster+pastries" target="_blank">
+      **beautiful toaster pastries**
+    </a>{' '}
+    and in the fulltext search from Trieve you'll still get{' '}
+    <a href="https://podfoods.trieve.ai/?term=Pod+Foods'+best+toaster+pastries" target="_blank">
+      **Pod Foods' best toaster pastries**
+    </a>
+    . That doesn't work in the current search implemantation.
+  </p>
+</div>
```
Comment (skeptrunedev) [inline src/content/post/podfoods-relevance-blog.mdx]: Same issue with dark mode

File: src/content/post/podfoods-relevance-blog.mdx [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+<div class="mx-10" style="font-size: 0.8em;">
+  Here are the current results:
+  <ol style="list-style-type: decimal;">
+    <li>❌ "UPGRADED kalumi BEAUTYfood marine collagen bars"</li>
+    <li>✅ a bar with raw honey from Honeymoon Chocolates</li>
+    <li>❌ a bar from Raw Rev. (Where the description says: **"Not a raw food."**)</li>
+    <li>❌ the "Raw Chocolate Essential Bars" from Rawmio, these are **vegan and do not contain raw honey**.</li>
+  </ol>
+  In Trieve, the top three results are all bars with raw honey.
+</div>
```
Comment (skeptrunedev) [inline src/content/post/podfoods-relevance-blog.mdx]: This can use markdown

File: src/content/post/podfoods-relevance-blog.mdx [hunk 4]
```diff
@@ -0,0 +1,300 @@
...
+  rightCaption="Trieve"
+  leftLink="https://podfoods.co/products?term=grapefruit%20weird%20yerba"
+  rightLink="https://podfoods.trieve.ai/?term=grapefruit%20weird%20yerba"
+/>
+
+### Specific Ingrediant + Form
+
+Here we focused on searching both the form of the product and a key ingredient. Quality results here also required a retrieval model that could look at the query as a whole, and not as disconnected aspects.
+
+<p style="margin-bottom: 0px;">Query: **raw honey bars**</p>
```
Comment (skeptrunedev) [inline src/content/post/podfoods-relevance-blog.mdx]: This can use markdown

File: src/content/post/podfoods-relevance-blog.mdx [hunk 5]
```diff
@@ -0,0 +1,300 @@
...
+
+<div class="mx-10" style="font-size: 0.8em;">
+  Here are the current results:
+  <ol style="list-style-type: decimal;">
+    <li>✅ The first result is from a brand based in Arizona.</li>
+    <li>❌ The third result is not based in Arizona (but does mention the Sonoran desert).</li>
+    <li>❌ The rest of the top results are from brands based in Washington, Oregon, California, and Virginia.</li>
+  </ol>
+  In Trieve, all of the top results are from brands based in Arizona.
+</div>
```
Comment (skeptrunedev) [inline src/content/post/podfoods-relevance-blog.mdx]: Same thing regarding markdown

File: src/content/post/podfoods-relevance-blog.mdx [hunk 6]
```diff
@@ -0,0 +1,300 @@
...
+</p>
+
+<small>
+  We were looking for a specific product, the{' '}
+  <a href="https://podfoods.co/raw-chocolate-essential-bars-rawmio-p10945.html?sku=26894-raw-chocolate-essential-bar-mint&i=765851">
+    Rawmio Essentials 70% Cacao Dark Chocolate Bar
+  </a>
+  , from which we constructed this keyword search query. In the current search, neither the particular bar nor anything
+  from the Rawmio brand appears in the top results.
+</small>
```
Comment (skeptrunedev) [inline src/content/post/podfoods-relevance-blog.mdx]: Markdown or maybe component for the small text

**Comment by danielsgriffin** [comment]
Added text to the Word to the Wise section:

> In e-commerce platforms, users often rely on keyword searches to find specific types of products. Some vector search techniques, while powerful, do not support keyword searches well because the terms are often outside the model's training data or domain.
> 
> If users are likely to use keyword searches and expect to find specific products, then the retrieval models should be selected (or adapted with rerankers) to support that use case. We address this issue with SPLADE in our fulltext search, and provide the flexibility for the searcher to also use the semantic search or the hybrid search.

I still need to:
- [ ] Add links to additional location search examples
- [ ] Fix cropping in main image (perhaps most straightforward for now is to manually make the uploaded image match the expected ratio):

image

**Comment by danielsgriffin** [comment]
More on the image display issue: The relevant files for the images are in src/components/blog/ListItem.astro (makes the items for the list of posts at /blog) and src/components/blog/SinglePost.astro (posts). The first renders 3:2 images and the second renders 16:9. Since the image in the PF blogpost is a search results screenshot, and is busy, these cropping / overflow differences look poor.

ListItem.astro
```
image && (
      <a class="relative block group" href={link ?? 'javascript:void(0)'}>
        <div class="relative h-0 pb-[56.25%] md:pb-[75%] md:h-72 lg:pb-[56.25%] overflow-hidden bg-gray-400 dark:bg-slate-700 rounded shadow-lg">
          {image && (
            <Image
              src={image}
              class="absolute inset-0 object-cover w-full h-full mb-6 rounded shadow-lg bg-gray-400 dark:bg-slate-700"
              widths={[400, 900]}
              width={900}
              sizes="(max-width: 900px) 400px, 900px"
              alt={post.title}
              aspectRatio="16:9"
              loading="lazy"
              decoding="async"
            />
          )}
        </div>
      </a>
    )
```

SinglePost.astro
```
post.image ? (
          <Image
            src={post.image}
            class="max-w-full lg:max-w-[900px] mx-auto mb-6 sm:rounded-md bg-gray-400 dark:bg-slate-700"
            widths={[400, 900]}
            sizes="(max-width: 900px) 400px, 900px"
            alt={post?.excerpt || ''}
            width={900}
            height={506}
            loading="eager"
            decoding="async"
          />
        ) : (
          <div class="max-w-3xl mx-auto px-4 sm:px-6">
            <div class="border-t dark:border-slate-700" />
          </div>
        )
```

Here is the issue when the image is designed to be 16x9:

image
image

File: src/content/post/podfoods-relevance-blog.mdx [hunk 7]
```diff
@@ -17,35 +17,34 @@ import QueryDiscussion from '~/components/common/QueryDiscussion.astro';
...
+Trieve focuses on helping our customers build great discovery experiences and [Pod Foods](https://podfoods.co/) is a wholesale grocery distribution company with a search platform for its B2B marketplace.
 
-Keyword search is enough for the vast majority of queries when someone knows what they're looking for, but less so when you're exploring. Pod Foods is really interesting because they are likely to have a significant number of users exploring rather than checking if a specific item is present. The whole point of the company is **to help retailers find new independent brands and help those brands be found**. So if discovery-focused search is not supported well, it's a big deal.
+Keyword search is ideal when you exactly what you're looking for, but the wrong fit for exploring. Pod Foods seemed like an ideal user-experience to us, because their explicit business goal is **to help retailers find new independent brands and help those brands be found**.
```
Comment (danielsgriffin) [inline src/content/post/podfoods-relevance-blog.mdx]: typo: when you know

File: src/content/post/podfoods-relevance-blog.mdx [hunk 8]
```diff
@@ -17,35 +17,34 @@ import QueryDiscussion from '~/components/common/QueryDiscussion.astro';
...
+Here we show several relevance comparisons between the existing search in Pod Foods and Trieve's AI-enhanced **fulltext** search.
 
 <TextBox>
-The demo also shows two other search types: **semantic**, a dense vector search using cosine distance vectors (here using [Jina english embeddings](https://jina.ai/news/jina-ai-launches-worlds-first-open-source-8k-text-embedding-rivaling-openai)), and **hybrid**, which takes in results from both fulltext and hybrid and re-ranks with a cross-encoder model (here the [BAAI/bge-reranker-large](https://huggingface.co/BAAI/bge-reranker-large)).
+  Our fulltext search uses **[SPLADE](https://github.com/naver/splade), a sparse neural search retrieval model**. There are also two other search types available: **semantic**, a dense vector search using cosine distance powered by [Jina english embeddings](https://jina.ai/news/jina-ai-launches-worlds-first-open-source-8k-text-embedding-rivaling-openai), and a **hybrid**, which takes in results from both fulltext and semantic then merges and ranks them with the [BAAI/bge-reranker-large](https://huggingface.co/BAAI/bge-reranker-large) cross-encoder model.
```
Comment (danielsgriffin) [inline src/content/post/podfoods-relevance-blog.mdx]: Do we want to say: "There are also two other search types available in the demo:"
Add: 'in the demo'

File: src/content/post/podfoods-relevance-blog.mdx [hunk 9]
```diff
@@ -149,24 +143,24 @@ Search results in Pod Foods currently are grouped as Product and each product pa
 
 ### Specific Ingredient + Form
 
-Here we focused on searching both the form of the product and a key ingredient. Quality results here also required a retrieval model that could look at the query as a whole, and not as disconnected aspects.
+Terms focused on general categories or nouns must be prioritized over descriptors for search to work correctly. We compare Trieve against the current search for queries containing both a type of product and key ingredient in this section.
 
 <QueryDiscussion query="raw honey bars">
-Here are the current results:
+  Rating of current results. Broadly, they are thrown off by the "honey" term to not focus on bars:
```
Comment (danielsgriffin) [inline src/content/post/podfoods-relevance-blog.mdx]: And thrown off by the Raw in the brand and product names.

**Comment by skeptrunedev** [comment]
Closing in favor of #27

**Pull request closed without merging**